lib/rubocop/cop/minitest/assert_raises_with_regexp_argument.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Checks for `assert_raises` with argument of regular expression literal.
+      # Arguments should be exceptions.
+      # The raised exception is returned and can be used
+      # to match against a regular expression.
+      #
+      # @example
+      #
+      #   # bad
+      #   assert_raises FooError, /some message/ do
+      #     obj.occur_error
+      #   end
+      #
+      #   # good
+      #   exception = assert_raises FooError do
+      #     obj.occur_error
+      #   end
+      #   assert_match(/some message/, exception.message)
+      #
+      class AssertRaisesWithRegexpArgument < Base
+        MSG = 'Do not pass regular expression literals to assert_raises. Test the resulting exception.'
+
+        def on_send(node)
+          return unless node.method?(:assert_raises)
+
+          return unless node.arguments.detect { |arg| regexp_literal?(arg) }
+
+          add_offense(node)
```

Comment:
```suggestion
          add_offense(node) if node.last_argument&.regexp_type?
```

Reply:
`assert_raises` accepts multiple arguments, but i think you're right...
the mistake we're trying to avoid springs from "expect this exception with this message" so checking the last arg is probably sufficient.